### PR TITLE
fix: split docs and start conceptual page

### PR DESCRIPTION
**Pull request by drbh**
This PR improves the guidance docs and adds a section that explains how grammars are applied on a technical level

### CODE REVIEW

File: docs/source/basic_tutorials/using_guidance.md [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+    json=data
+)
+print(response.json())
+# {'generated_text': '{ "activity": "bike riding", "animals": ["puppy","cat","raccoon"],"animals_seen": 3, "location":"park" }'}
+
+```
+
+### JSON Schema Integration
+
+If Pydantic's not your style, go raw with direct JSON Schema integration. This is simliar to the first example but with programmatic control.
```
Comment (aymeric-roucher) [inline docs/source/basic_tutorials/using_guidance.md:125]: Typo: "simliar"

File: docs/source/conceptual/guidance.md [hunk 2]
```diff
@@ -1,425 +1,64 @@
...
+3. The process of choosing one of those tokens is called `sampling`. The model samples from the distribution of probabilities to choose the next token. In TGI all of the steps before sampling are called `processor`. Grammars are applied as a processor that masks out tokens that are not allowed by the grammar.
 
-### The Grammar Parameter
+<div class="flex justify-center">
+    <img
+        class="block dark:hidden"
+        src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/tgi/logit-grammar-mask.gif"
+    />
+    <img
+        class="hidden dark:block"
```
Comment (aymeric-roucher) [inline docs/source/conceptual/guidance.md:44]: These gifs are great! I also liked this [high-level graph](https://huggingface.slack.com/archives/C05CFK1HM0T/p1714157053184219?thread_ts=1714052150.677459&cid=C05CFK1HM0T) on slack

File: docs/source/basic_tutorials/using_guidance.md (hunk 1 above)
Comment (drbh) [reply docs/source/basic_tutorials/using_guidance.md:125]: thanks!

File: docs/source/conceptual/guidance.md (hunk 2 above)
Comment (drbh) [reply docs/source/conceptual/guidance.md:55]: @aymeric-roucher **updated and remade the other steps of the diagram as gifs. Please let me know if its missing anything needed from the original

**Review by Narsil — APPROVED** [review]

**Pull request merged**